### PR TITLE
Fix replication controller usage in kube e2e tests

### PR DESCRIPTION
**Pull request by marun**
Some kube e2e tests (e.g. the one for NodePort services) use replication
controllers to launch containers that require anyuid.  To enable this,
the test project service accounts need to be added to the anyuid scc.

### CODE REVIEW

**Comment by pweil-** [comment]
If an e2e test is introduced that is trying to create privileged pods via a replication controller it will still fail.  We could avoid that by prioritizing the privileged SCC for kube e2e, then we wouldn't need to add the SA to the anyuid SCC.

@liggitt - thoughts on that?

**Comment by pweil-** [comment]
Actually, disregard.  If the pod is requesting privileged access then it will have parameters set (pids, net, whatever) that would trigger the use of the privileged SCC for the SA.  This was just because of the RunAsUser.  

:+1:

File: test/extended/util/test.go [hunk 1]
```diff
@@ -96,31 +98,41 @@ func ensureKubeE2EPrivilegedSA() {
...
 		namespaces, err := c.Namespaces().List(labels.Everything(), fields.Everything())
 		if err != nil {
 			FatalErr(err)
 		}
-		groups := []string{}
-		for _, name := range priv.Groups {
-			if !strings.Contains(name, "e2e-") {
-				groups = append(groups, name)
-			}
+		addE2EServiceAccountsToSCC(c, namespaces, "privileged")
```
Comment (liggitt) [inline test/extended/util/test.go]: better comments:
```
// add to the "privileged" scc to ensure pods that explicitly request extra capabilities are not rejected
addE2EServiceAccountsToSCC(c, namespaces, "privileged")
// add to the "anyuid" scc to ensure pods that don't specify a uid don't get forced into a range (mimics upstream behavior)
addE2EServiceAccountsToSCC(c, namespaces, "anyuid")
```

**Comment by liggitt** [comment]
nit on comments, LGTM otherwise

File: test/extended/util/test.go (hunk 1 above)
Comment (marun) [inline test/extended/util/test.go]: Done

**Comment by liggitt** [comment]
[test][extended:core]

**Comment by dcbw** [comment]
LGTM from 10,000ft...

**Comment by liggitt** [comment]
actually, can you remove the things that now work from the SKIP_TESTS list in extended/core.sh and we'll re-run an extended core test?

**Comment by marun** [comment]
@liggitt Ok, will do

**Comment by marun** [comment]
@liggitt I can't see any core skipped tests that would pass with only this change.  'Addon update' does use replication controllers, but it also requires ssh which won't work for a single-host test.  'Networking should function for intra-pod' uses a replication controller, but also won't work for a single-host deployment.  The remainder of the skipped tests appear to have documented reasons for failure that preclude unskipping at this time.

**Comment by liggitt** [comment]
ok, re-[test]

**Comment by marun** [comment]
[test]

Failure doesn't appear related to this PR and may be a flake.

**Comment by marun** [comment]
[testonlyextended][extended:core]

**Comment by marun** [comment]
@liggitt Can you please remove your test directives?  I'd like to run just the extended tests (i.e. [testonlyextended][extended:core]) since this change doesn't affect the non-extended tests.

**Comment by marun** [comment]
@liggitt I think this is ready to merge.

**Comment by liggitt** [comment]
[merge]

**Comment by marun** [comment]
@liggitt re-kick?

**Comment by deads2k** [comment]
re[merge]

**Comment by deads2k** [comment]
re[test]

**Pull request merged**